test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala

```diff
@@ -0,0 +1,1328 @@
+
+      "update of draft fails" in {
+
+        def performAction(): Future[Result] = controller.supportingEvidenceExpiredSubmit()(FakeRequest())
+
+        val supportingEvidence = sample[SupportingEvidence]
+
+        val answers = IncompleteSupportingEvidenceAnswers(
+          doYouWantToUploadSupportingEvidence = Some(true),
+          evidences                           = List.fill(1)(supportingEvidence),
+          expiredEvidences                    = List(supportingEvidence)
+        )
+
+        val (session, journey, draftReturn) = sessionWithSingleDisposalState(Some(answers))
+
+        val updatedDraftReturn =
+          draftReturn.copy(supportingEvidenceAnswers =
+            Some(
+              answers.copy(
+                doYouWantToUploadSupportingEvidence = Some(true),
+                expiredEvidences                    = List.empty
+              )
+            )
+          )
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+          mockStoreDraftReturn(
+            updatedDraftReturn,
+            journey.subscribedDetails.cgtReference,
+            journey.agentReferenceNumber
+          )(Left(Error("update failed")))
+        }
+
+        checkIsTechnicalErrorPage(
+          performAction()
+        )
+      }
+    }
+
+    "redirect to check your answers" when {
+
+      "if answers are complete" in {
+        def performAction(): Future[Result] = controller.supportingEvidenceExpired()(FakeRequest())
+
+        val answers = CompleteSupportingEvidenceAnswers(
+          doYouWantToUploadSupportingEvidence = false,
+          evidences                           = List.empty
+        )
+
+        val (session, _, _) = sessionWithSingleDisposalState(Some(answers))
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkIsRedirect(
+          performAction(),
+          routes.SupportingEvidenceController.checkYourAnswers()
+        )
+      }
+    }
+
+    "display supporting evidence expired page" when {
+
+      "supporting evidence has expired" in {
+        def performAction(): Future[Result] = controller.supportingEvidenceExpired()(FakeRequest())
+
+        val supportingEvidence = sample[SupportingEvidence]
+
+        val answers = IncompleteSupportingEvidenceAnswers(
+          doYouWantToUploadSupportingEvidence = Some(true),
+          evidences                           = List.fill(0)(supportingEvidence),
+          expiredEvidences                    = List(supportingEvidence)
+        )
+
+        val (session, _, _) = sessionWithSingleDisposalState(Some(answers))
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey("supporting-evidence.expired.title")
+        )
+      }
+    }
+
+    "redirect to check your answers " when {
```

Comment:
it'd be clearer if the submit endpoint of the expired page had it's own section in this spec outside of the display page endpoint